p2pserver/message/utils/msg_handler.go

```diff
@@ -259,12 +262,14 @@ func VersionHandle(data *msgCommon.MsgPayload, p2p p2p.P2P, pid *evtActor.PID, a
 			log.Warn("The node handshake with itself")
 			remotePeer.CloseCons()
 			p2p.RemovePeerConsAddress(data.Addr)
+			p2p.RemoveFromConnectingList(data.Addr)
 			return errors.New("The node handshake with itself ")
 		}
 
 		s := remotePeer.GetConsState()
 		if s != msgCommon.INIT && s != msgCommon.HAND {
 			log.Warn("Unknown status to received version", s)
+			p2p.RemoveFromConnectingList(data.Addr)
```

Comment:
you may also need close the net connection .

Reply:
clean remove logic, close connection contain same action
if dual port not support, set consensus port to zero which avoid be connected by other peer